### PR TITLE
Declare mock SMM Access2 Protocol.

### PR DESCRIPTION
**Pull request by ishih1**
## Description

Declare mock SMM Access2 Protocol for google test.

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Run the google test.

## Integration Instructions

N/A

### CODE REVIEW

**Comment by ishih1** [comment]
> @ishih1 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="{lenovo}"]

**Comment by ishih1** [comment]
> @microsoft-github-policy-service agree company="your company"

@microsoft-github-policy-service agree company="Lenovo"

**Comment by ishih1** [comment]
@VivianNK 
Please review this PR, thanks.

**Comment by VivianNK** [comment]
> @VivianNK Please review this PR, thanks.

@ishih1 Hi, please create a PR to edk2 (https://github.com/tianocore/edk2) with this change

**Comment by ishih1** [comment]
> > @VivianNK Please review this PR, thanks.
> 
> @ishih1 Hi, please create a PR to edk2 (https://github.com/tianocore/edk2) with this change

The PR has already been created. https://github.com/tianocore/edk2/pull/6299

**Comment by ishih1** [comment]
@VivianNK 
Please approve the PR if there is no more comment.

**Comment by ishih1** [comment]
@VivianNK 
Please help to merge this PR. Thanks.

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/** @file
```
Comment (VivianNK) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h]: Please add the filename here

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  It abstracts the location and characteristics of SMRAM.  The expectation is
+  that the north bridge or memory controller would publish this protocol.
+
+  The principal functionality found in the memory controller includes the following:
+  - Exposing the SMRAM to all non-SMM agents, or the "open" state
+  - Shrouding the SMRAM to all but the SMM agents, or the "closed" state
+  - Preserving the system integrity, or "locking" the SMRAM, such that the settings cannot be
+    perturbed by either boot service or runtime agents
+
+  Copyright (c) 2009 - 2018, Intel Corporation. All rights reserved.<BR>
```
Comment (VivianNK) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h]: Please add Microsoft copyright

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h (hunk 1 above)
Comment (ishih1) [reply MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h]: Please check the update.

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h (hunk 2 above)
Comment (ishih1) [reply MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmAccess2.h]: Please check the update.

**Comment by ishih1** [comment]
@VivianNK 
Please check the update.

**Review by VivianNK — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

**Pull request merged**